sklearn/ensemble/_stacking.py

```diff
@@ -104,11 +104,25 @@ def _concatenate_predictions(self, X, predictions):
         """
         X_meta = []
         for est_idx, preds in enumerate(predictions):
-            # case where the estimator returned a 1D array
-            if preds.ndim == 1:
+            # Some estimator return a 1D array for predictions
+            # which must be 2-dimensional arrays.
+            if hasattr(preds, "ndim") and preds.ndim == 1:
                 X_meta.append(preds.reshape(-1, 1))
             else:
                 if (
+                    self.stack_method_[est_idx] == "predict_proba"
+                    and self._type_of_target == "multilabel-indicator"
+                    and isinstance(preds, list)
+                ):
+                    # `preds` is here a list of n_targets 2D ndarrays of 2 columns.
+                    # The two columns contains the probabilities of the negative
+                    # and positive classes respectively.
+                    # For binary classification, we only work with probabilities
+                    # of the positive class (the second column), hence we drop
+                    # the first column.
+                    for pred in preds:
+                        X_meta.append(pred[:, 1:])
+                elif (
```

Comment:
Coming back to this case: potentially we might not treat a binary classification problem in this branch (the next branch does, and it's somewhat explicit in its test).

Still, the representation might also work:
```suggestion
                    self.stack_method_[est_idx] == "predict_proba"
                    and self._type_of_target == "multilabel-indicator"
                    and isinstance(preds, list)
                ):
                    # `preds` is here a list of n_targets 2D ndarrays of 
                    # `n_classes` columns. The k-th column contains the
                    # probabilities of the samples belonging the k-th class.
                    #
                    # Since those probabilities must sum to one for each sample,
                    # we can work with probabilities of `n_classes - 1` classes.
                    # Hence we drop the first column.
                    for pred in preds:
                        X_meta.append(pred[:, 1:])
                elif (
```

:warning:  I am knowledgeable enough about the various representations and the way they are handled and identified. Thus I would wait for someone else to confirm before accepting this suggestion.

Reply:
Ok, thanks for your feedback :)
I got your point and correct me if I wrong, but the context of this condition is `"multilabel-indicator"`, which is a binary problem for each column of the target y.
In addition, `preds` being probabilities from the `predict_proba` method alleviates the ambiguity for this branch, in my understanding.
Who would you suggest for reviewing this part?